tokio-util/src/codec/length_delimited.rs

```diff
@@ -546,12 +543,15 @@ impl Decoder for LengthDelimitedCodec {
     }
 }
 
-impl Encoder for LengthDelimitedCodec {
-    type Item = Bytes;
+impl<T> Encoder<T> for LengthDelimitedCodec
+where
+    T: AsRef<[u8]>,
```

Comment:
Same here I am not following why moving to asref for something like these codecs is useful? I understand why we want to trait to accept a generic but not following the advantage here.

https://docs.rs/bytes/0.5.4/bytes/trait.Buf.html#impl-Buf-for-%26%27_%20%5Bu8%5D we also already have a `impl Buf for &[u8]`, so changing this to buf should support the special case of using `Bytes` but also accepting a reference via asref?

Reply:
It was a concrete `Bytes` before and now it could be generic over `AsRef<[u8]>` or `Buf`. The generated code will be equivalent either way but the implementation only requires `as_ref` (not the other methods on `Buf` so I figured `AsRef` were more general.

Reply:
Reverted to `Bytes` for now.